src/transformers/modeling_utils.py

```diff
@@ -4008,8 +4019,12 @@ def save_pretrained(
             repo_id = self._create_repo(repo_id, **kwargs)
             files_timestamps = self._get_files_timestamps(save_directory)
 
+        metadata = {}
         if hf_quantizer is not None:
             state_dict = hf_quantizer.get_state_dict(self)
+            if isinstance(state_dict, tuple):
+                state_dict, metadata = state_dict
```

Comment:
this might be a bit confusing I feel, I think we should use a different API to get tensor_data_dict and metadata